### PR TITLE
Improvements to the displayable image view so that it properly cancel…

### PR DESCRIPTION
**Pull request by MartinJibber**
…s outstanding network requests.

### CODE REVIEW

File: Shared/UI/Views/DisplayableImageView.swift [hunk 1]
```diff
@@ -113,13 +113,15 @@ class DisplayableImageView: View {
         if let photo = displayable.image {
             self.showResult(for: photo)
         } else if let url = displayable.url {
-            self.downloadAndSetImage(url: url)
+            Task {
+                await self.downloadAndSetImage(url: url)
+            }.add(to: self.taskPool)
```
Comment (MartinJibber) [inline Shared/UI/Views/DisplayableImageView.swift:118]: Making downloads cancellable.

File: Shared/UI/Views/DisplayableImageView.swift [hunk 2]
```diff
@@ -138,8 +140,9 @@ class DisplayableImageView: View {
     }
 
     @MainActor
-    private func downloadAndSetImage(url: URL) {
-        self.imageView.sd_setImage(with: url, placeholderImage: nil, options: [.refreshCached]) { received, expected, url in
+    private func downloadAndSetImage(url: URL) async {
```
Comment (MartinJibber) [inline Shared/UI/Views/DisplayableImageView.swift:143]: Converting this to an async function to make it easier to cancel.

File: Shared/UI/Views/DisplayableImageView.swift [hunk 3]
```diff
@@ -149,14 +152,11 @@ class DisplayableImageView: View {
...
-        } completed: { [weak self] image, error, cacheType, url in
-            guard let `self` = self, let downloadedImage = image else {
-                self?.showResult(for: nil)
-                return
-            }
+        })
 
-            self.showResult(for: downloadedImage)
-        }
+        guard !Task.isCancelled else { return }
```
Comment (MartinJibber) [inline Shared/UI/Views/DisplayableImageView.swift:157]: If the task was cancelled in the middle of the download, then don't update the image.

File: Shared/UI/Views/DisplayableImageView.swift [hunk 4]
```diff
@@ -188,6 +188,10 @@ class DisplayableImageView: View {
             cancellable.cancel()
         }
 
+        Task {
+            await self.taskPool.cancelAndRemoveAll()
+        }
```
Comment (MartinJibber) [inline Shared/UI/Views/DisplayableImageView.swift:193]: Cancel all tasks when reset is called.

File: Shared/UI/Views/DisplayableImageView.swift [hunk 5]
```diff
@@ -198,3 +202,23 @@ class DisplayableImageView: View {
...
+                    continuation.resume(throwing: error)
+                }
+
+                continuation.resume(returning: image)
+            }
+        }
+
+        return image
+    }
+}
```
Comment (MartinJibber) [inline Shared/UI/Views/DisplayableImageView.swift:224]: An async version of the sd web image extension.

**Review by Benjamindodgson — APPROVED** [review]
Wow this async stuff is helpful for this type of issue

**Pull request merged**